File: examples/typescript-demo/math.ts
/**
 * A simple math utility.
 * This file is used as a demo for `mcpify`.
 *
 * Run: npx mcpify generate ./math.ts
 */

/**
 * Adds two numbers together.
 * @param a - First operand
 * @param b - Second operand
 */
export function add(a: number, b: number): number {
  return a + b;
}

/**
 * Subtracts b from a.
 * @param a - Value to subtract from
 * @param b - Value to subtract
 */
export function subtract(a: number, b: number): number {
  return a - b;
}

/**
 * Converts a temperature from Fahrenheit to Celsius.
 * @param fahrenheit - Temperature in degrees Fahrenheit
 */
export function fahrenheitToCelsius(fahrenheit: number): number {
  return ((fahrenheit - 32) * 5) / 9;
}

/**
 * Greets a user by name.
 * @param name - The user's name
 * @param loud - Whether to shout the greeting
 */
export function greet(name: string, loud?: boolean): string {
  const msg = `Hello, ${name}!`;
  return loud ? msg.toUpperCase() : msg;
}

File: examples/python-demo/math.py
"""
A simple math utility.
This file is used as a demo for mcpify.

Run: npx mcpify generate ./math.py
"""


def add(a: int, b: int) -> int:
    """Add two integers together.

    Args:
        a (int): First operand.
        b (int): Second operand.
    """
    return a + b


def subtract(a: int, b: int) -> int:
    """Subtract b from a.

    Args:
        a (int): Value to subtract from.
        b (int): Value to subtract.
    """
    return a - b


def fahrenheit_to_celsius(fahrenheit: float) -> float:
    """Convert a temperature from Fahrenheit to Celsius.

    Args:
        fahrenheit (float): Temperature in degrees Fahrenheit.
    """
    return (fahrenheit - 32) * 5 / 9


def greet(name: str, loud: bool = False) -> str:
    """Greet a user by name.

    Args:
        name (str): The user's name.
        loud (bool): Whether to shout the greeting.
    """
    msg = f"Hello, {name}!"
    return msg.upper() if loud else msg
